verilog/CST/type.h

```diff
@@ -33,6 +33,11 @@ std::vector<verible::TreeSearchMatch> FindAllDataTypeDeclarations(
 std::vector<verible::TreeSearchMatch> FindAllTypeDeclarations(
     const verible::Symbol&);
 
+// Finds all node kEnumDataType declarations. Used for testing if the type
+// declaration is an enum.
+std::vector<verible::TreeSearchMatch> FindAllEnumDataTypeDeclarations(
```

Comment:
Technically, not all appearances of the enum construct are type declarations, as some are anonymous in data declarations.  (It just happens that you're only using it inside type declarations at this time.)
Maybe name this `FindAllEnumDeclarations`?

Reply:
You're right that they can be anonymous, but here I was going by node name (`kEnumDataType`) in `verilog_nonterminals.h`.
I'm open to renaming this, but then I suppose the `FindAllStructDataTypeDeclarations` and `FindAllUnionDataTypeDeclarations` should be renamed as well?

Reply:
Also good point.  I agree that these cases of Struct/Union/Enum should be consistent with each other as well.  If it's not too much trouble, could we change those as well?  `FindAllStructDeclarations` and `FindAllEnumDeclarations`.  This way, we can use `TypeDeclarations` to explicitly refer to `typedef` constructs.

Reply:
Following your proposition, I renamed them to `kEnumType`, `kStructType`, `kUnionType`.